src/client/containers/CartPage/CartPage.Container.js

```diff
@@ -1,48 +1,77 @@
+/* eslint-disable react-hooks/rules-of-hooks */
 import React from 'react';
 import './CartPage.Style.css';
-import { useParams } from 'react-router-dom';
 import Loader from '../../components/Loader/Loader.component';
 import ShoppingItem from '../../components/ShoppingItem/ShoppingItem';
-import DeliveryInfo from '../../components/DeliveryInfo/DeliveryInfo.component';
 import TotalPrice from '../../components/TotalPriceCard/TotalPriceCard.component';
 import ContactForm from '../../components/ContactForm/ContactForm.component';
 import ButtonComponent from '../../components/Button/Button.component';
-import { useFetchApi } from '../../hooks/UseFetchApi';
 import { useFirebase } from '../../firebase/FirebaseContext';
 import { PropTypes } from 'prop-types';
+import { useShoppingCartContext } from '../../context/shoppingCart/shoppingCartContext';
+import { useHistory } from 'react-router-dom';
+import DeliveryInfoV2 from '../../components/DeliveryInfo/DeliveryInfoV2.component';
+import { useFetchApi } from '../../hooks/UseFetchApi';
 
 const CartPageContainer = ({ isAuthenticated }) => {
   const [cartItem, setCartItem] = React.useState([]);
-  const [user, setUser] = React.useState({});
-  const [userId, setUserId] = React.useState('');
+  const [IsLoading, setIsLoading] = React.useState(true);
   const [total, setTotal] = React.useState(0);
-  const [itemCost, setItemCost] = React.useState([]);
-  const { id } = useParams();
+
+  const [user, setUser] = React.useState({});
   const { auth } = useFirebase();
 
-  const orderData = useFetchApi(`orders/${id}`);
+  const { shoppingCart, changeProductQuantity } = useShoppingCartContext();
+  const history = useHistory();
+
+  React.useEffect(() => {
+    Promise.all(
+      Object.keys(shoppingCart).map((productId) => {
+        return fetch(`api/products/${productId}`)
+          .then((res) => res.json())
+          .then((data) => {
+            const newsItem = {
+              // eslint-disable-next-line object-shorthand
+              productId: productId,
+              quantity: shoppingCart[productId],
+              name: data[0].name,
+              picture: data[0].picture,
+              price: data[0].price,
+              // eslint-disable-next-line @typescript-eslint/camelcase
+              stock_amount: data[0].stock_amount,
+            };
+            return newsItem;
+          });
+      }),
+    ).then((data) => {
+      setCartItem(data);
+      setIsLoading(false);
+    });
+  }, [shoppingCart]);
 
   React.useEffect(() => {
     let totalCost = 0;
     // eslint-disable-next-line no-plusplus
-    for (let i = 0; i < itemCost.length; i++) {
-      totalCost += itemCost[i];
+    for (let i = 0; i < cartItem.length; i++) {
+      // eslint-disable-next-line operator-assignment
+      totalCost = totalCost + cartItem[i].price * cartItem[i].quantity;
```

Comment:
If you want to write it a more declarative way you could also do this:
```suggestion
    const totalCost = cartItem.reduce((acc, cartItem) =>  acc += cartItem.price * cartItem.quantity, 0);
```

Reply:
thanks.. will do the changes

Reply:
done